src/features/authentication/Auth.tsx

```diff
@@ -148,6 +210,26 @@ function Auth() {
     const rpc = new RPC(provider);
     return await rpc.getPrivateKey();
   };
+  
+async function buildAndConnectClient(_wallet?: EthereumWallet) {
+  const wallet = _wallet ?? EthereumWallet.createRandom();
+  const provider = new EthereumProvider(wallet);
+  const storageProvider = new CustomStorageProvider();
+  const account = new AccountId({
+    address: wallet.address,
+    chainId: `eip155:1`,
+  });
+  const client = new LigoClient(
+    provider,
```

Comment:
Instead of creating the custom `EthereumProvider`, you should be able to pass the `SafeEventEmitterProvider` from Web3Auth directly here.